instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -83,9 +91,16 @@ def _otel_client_attributes
 
             attributes[SemanticConventions::Trace::DB_NAME] = _otel_database_name
             attributes[SemanticConventions::Trace::PEER_SERVICE] = config[:peer_service]
+
             attributes
           end
 
+          def db_sql_table_name(sql)
+            Regexp.last_match(1) if sql =~ TABLE_NAME
+          rescue StandardError
+            nil
```

Comment:
If an error occurs and the attribute is set to `nil`, then the SDK will report an error. It would be best to avoid setting any invalid attributes. 

Please ensure that the attribute is not set in cases where the table name could not be extracted. 

https://github.com/open-telemetry/opentelemetry-ruby/blob/555b062ef9421784c132aa9b97b29ec637b13b0f/sdk/lib/opentelemetry/sdk/trace/span.rb#L277

https://github.com/open-telemetry/opentelemetry-ruby/blob/555b062ef9421784c132aa9b97b29ec637b13b0f/sdk/lib/opentelemetry/sdk/internal.rb#L57

Reply:
Made some changes! Will log an error if trouble getting the table name and won't report. Is the error logging here okay?